syntax = "proto3";

package productlookup;

service ProductSearch {
  rpc SearchProduct(SearchProductRequest) returns (SearchProductResponse) {}
}

message SearchProductRequest {
  string query = 1;
}

message ProductData {
  string sku_id = 1;
  string product_name = 2;
  string brand = 3;
  string description = 4;
  string price = 5;
  string product_url = 6;
  string part_number = 7;
}

message SearchProductResponse {
  repeated ProductData products = 1;
}